hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFSWithObjectStoreCreate.java

```diff
@@ -209,6 +215,94 @@ public void testObjectStoreCreateWithO3fs() throws Exception {
 
   }
 
+
+  @Test
+  public void testKeyCreationFailDuetoDirectoryCreationBeforeCommit()
+      throws Exception {
+    OzoneVolume ozoneVolume =
+        cluster.getRpcClient().getObjectStore().getVolume(volumeName);
+
+    OzoneBucket ozoneBucket = ozoneVolume.getBucket(bucketName);
+
+    OzoneOutputStream ozoneOutputStream =
+        ozoneBucket.createKey("/a/b/c", 10);
+    byte[] b = new byte[10];
+    Arrays.fill(b, (byte)96);
+    ozoneOutputStream.write(b);
+
+    // Before close create directory with same name.
+    o3fs.mkdirs(new Path("/a/b/c"));
```

Comment:
Do we have a test case that tests the reverse scenario? i.e. create the dir first, then try to create a file with the same name. Also are we testing this via the object store interface with the flag enabled?

Reply:
>Do we have a test case that tests the reverse scenario? i.e. create the dir first, then try to create a file with the same >name
No such test is currently available. Will post shortly this test also.

>Also are we testing this via the object store interface with the flag enabled?
Yes test is with flag enabled.

Reply:
Added the test.